File: oop2.java
class dog{
    // this are attributes
    private String name;
    private int age;
    private String dateAttended;


    //creating constructor same as __init__ in python
    public dog(String name, int age, String dateAttended){
        this.name = name;
        this.age = age;
        this.dateAttended = dateAttended;
    }

    // java method for displaying info
    public void displayinfoofdog(){
        System.out.println("name:" + name+",age:"+ age+",date attended: dateattended");
    }

    public static void main(String[] args){
        dog dog1 = dog("tommy",'2022-09-09')
        dog dog2 = dog("jerry",4,"2087-05-04")
    }

    dog1.displayinfoofdog();
    dog2.displayinfoofdog();
}

File: oop2.py
'''Imagine running a dog exhibition camp with hundreds of dogs, and you have to keep track of names,ages and date attended for each dog.
How would you desgin simple resuable model for the dogs?'''

class dog:
    def __init__(self,name,age,date):
        self.name = name
        self.age = age
        self.date =date
    
    def displayinfoofdog(self):
        print(f"name: {self.name}, age: {self.age}, date attended: {self.date}")


dog1 = dog('tommy',2,'2025-05-31')
dog2 = dog('jerry',4,'2026-06-98')

dog1.displayinfoofdog()
dog2.displayinfoofdog()


